core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -893,14 +893,16 @@ object ZLayer {
 
   /**
    * Constructs a layer that purely depends on the specified service, which
-   * must return one or more services.
+   * must return one or more services. For the more common variant that returns
+   * a single service see `fromService`.
    */
   def fromServiceMany[A: Tagged, B <: Has[_]](f: A => B): ZLayer[Has[A], Nothing, B] =
     fromServiceManyM[A, Any, Nothing, B](a => ZIO.succeedNow(f(a)))
 
   /**
-   * Constructs a layer that purely depends on the specified services, which
-   * must return one or more services.
+   * Constructs a layer that purely depends on the specified service, which
```

Comment:
I would use plural for the "services" (n > 1) cases.

Reply:
Good catch! Fixed now.